mindmeld/models/embedder_models.py

```diff
@@ -757,10 +758,14 @@ class GloveEmbedder(Embedder):
 
     def __init__(self, app_path, **kwargs):
         super().__init__(app_path, **kwargs)
-        self.glove_tokenizer = Tokenizer()
+        self.glove_text_preparation_pipeline = (
+            TextPreparationPipelineFactory.create_default_text_preparation_pipeline()
+        )
 
     def tokenize(self, text):
-        tokens = self.glove_tokenizer.tokenize(text, keep_special_chars=False)
+        tokens = self.glove_text_preparation_pipeline.get_normalized_tokens_from_text(
```

Comment:
Are we making sure `keep_special_chars ` is being propagated correctly?

Reply:
Good point, will take a closer look to double check

Reply:
Many of the regex rules remove "Special Characters" that can be found in between words or numbers. Here is an example rule:

```
    "ReplacePunctuationAtWordEndWithSpace": RegexNormalizerRule(
        pattern=r"[^\w\d&" + CURRENCY_SYMBOLS + EXCEPTION_CHARS + r"]+$",
        replacement=" ",
    )
```

The EXCEPTION_CHARS currently refers to `r"\@\[\]\|\{\}'"`. When keep_special_chars (old code) is used, these EXCEPTION_CHARS are applied such that these special chars are not removed.

I think we may be able to skip the "{" "}" characters since regex detects MindMeld annotation patterns now. Wonder if we could skip "[" and "]" as well? In which case are brackets used.

Reply:
I don't think we skip `[`/`]` with the current changes in the PR, no?

Reply:
Added support for this from the config as discussed.